ale/drivers/lro_drivers.py

```diff
@@ -1041,12 +1091,111 @@ def sampling_factor(self):
 
     @property
     def num_frames(self):
+        """
+        Number of frames in the image
+
+        Returns
+        -------
+        : int
+          Number of frames in the image
+        """
         return self.image_lines // (self.framelet_height // self.sampling_factor)
 
 
     @property
     def framelet_height(self):
+        """
+        Return the number of lines in a framelet.
+
+        Returns
+        -------
+        : int
+          The number of lines in a framelet
+        """
         if self.instrument_id == "LRO_LROCWAC_UV":
             return 16
         elif self.instrument_id == "LRO_LROCWAC_VIS":
             return 14
+
+
+    @property
+    def filter_number(self):
+        """
+        Return the filter number from the cub label
+
+        Returns
+        -------
+        : int
+          The filter number
+        """
+        try:
+            return self.label['IsisCube']['BandBin']['FilterNumber'][0]
+        except:
+            return self.label['IsisCube']['BandBin']['FilterNumber']
+
+
+    @property
+    def fikid(self):
+        """
+        Naif ID code of the filter dependent instrument codes.
+
+        Expects ikid to be defined. This should be the integer Naif ID code for
+        the instrument.
+
+        Returns
+        -------
+        : int
+          Naif ID code used in calculating focal length
+        """
+        if self.instrument_id == "LRO_LROCWAC_UV":
+            base = -85640
+        elif self.instrument_id == "LRO_LROCWAC_VIS":
+            base = -85630
+
+        fikid = base - self.filter_number
+        return fikid
+
+
+    @property
+    def pixel2focal_x(self):
+        """
+        Expects fikid to be defined. This must be the integer Naif id code of the filter
+
+        Returns
+        -------
+        : list<double>
+        detector to focal plane x
+        """
+        return list(spice.gdpool('INS{}_TRANSX'.format(self.fikid), 0, 3))
+
+
+    @property
+    def pixel2focal_y(self):
+        """
+        Expects fikid to be defined. This must be the integer Naif id code of the filter
+
+        Returns
+        -------
+        : list<double>
+        detector to focal plane y
+        """
+        return list(spice.gdpool('INS{}_TRANSY'.format(self.fikid), 0, 3))
+
+
+    @property
+    def detector_start_line(self):
+        """
+        Filter-specific starting line
+
+        Returns
+        -------
+        : int
+          Zero based Detector line corresponding to the first image line
+        """
+        offset = list(spice.gdpool('INS{}_FILTER_OFFSET'.format(self.fikid), 0, 3))
+        try:
+            # If multiple items are present, use the first one
+            offset = offset[0]
```

Comment:
We need to take the one that corresponds to the Band ID. Here's what the SPICE looks like:

```
INS-85621_FILTER_OFFSET                   = (703, 728, 754, 781, 806)
 INS-85621_FILTER_BANDID                   = (3, 4, 5, 6, 7)
```

So for an image where the first band has ID 6, we would want the 4th element of the offset array.

Reply:
-85621 (VIS) and -85626 (UV) are marked as deprecated in the IK, and the filter-specific IKIDs are used instead.  Each filter defines its own int offset that's accessible with INS{FIKID}_FILTER_OFFSET instead of having to maintain a mapping relative to the base IKID.

If the first band has ID 6, we're getting FIKID -85634, which should be listed under NaifKeywords as INS-85634_FILTER_OFFSET = 781